builtin/game/chat.lua

```diff
@@ -1034,12 +1034,11 @@ core.register_chatcommand("time", {
 		end
 		local hour, minute = param:match("^(%d+):(%d+)$")
 		if not hour then
-			local new_time = tonumber(param)
-			if not new_time then
-				return false, S("Invalid time.")
+			local new_time = tonumber(param) or -1
+			if new_time ~= new_time or new_time < 0 or new_time > 24000 then
```

Comment:
```new_time ~= new_time``` is always false and can be left out:
```lua
if new_time < 0 or new_time > 24000 then
```
since you already check for nil in the line above and adjust to -1

Reply:
weird, this is not referencing the line I meant